src/gui/widgets/toggle_button.cpp

```diff
@@ -121,7 +121,15 @@ void toggle_button::set_value(const unsigned selected)
 
 	// Check for get_window() is here to prevent the callback from
 	// being called when the initial value is set.
-	if(callback_state_change_ && get_window() != nullptr) {
+	//
+	// TODO: see if this is necessary with the signal handler.
```

Comment:
As the author of the check: yes, it remains necessary with the signal handler. You may remove the TODO.